src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java

```diff
@@ -91,14 +91,14 @@ public static void removeProvider(final IStandardRequestManager manager, final I
      * @throws IllegalArgumentException is thrown when the token is not registered to a provider, or when the data stored in the manager is in conflict.
      */
     @SuppressWarnings(Suppression.UNCHECKED)
-    public static void removeProviderInternal(final IStandardRequestManager manager, final IToken token) throws IllegalArgumentException
+    public static void removeProviderInternal(final IStandardRequestManager manager, final IToken<?> token) throws IllegalArgumentException
     {
         final IRequestResolverProvider provider = getProvider(manager, token);
 
         LogHandler.log("Removing provider: " + provider);
 
         //Get the resolvers that are being removed.
-        final ImmutableCollection<IToken> assignedResolvers = getRegisteredResolvers(manager, token);
+        final ImmutableCollection<IToken<?>> assignedResolvers = getRegisteredResolvers(manager, token);
         for (final IToken resolverToken : assignedResolvers)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS135)